tasks/idp.yml

```diff
@@ -0,0 +1,914 @@
+    backup: yes
+
+- name: 'Set intercept/consent-intercept-config.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/consent-intercept-config.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/consent-intercept-config.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/context-check-intercept-config.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/context-check-intercept-config.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/context-check-intercept-config.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/expiring-password-intercept-config.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/expiring-password-intercept-config.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/expiring-password-intercept-config.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/external-intercept-config.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/external-intercept-config.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/external-intercept-config.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/impersonate-intercept-config.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/impersonate-intercept-config.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/impersonate-intercept-config.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/intercept-events-flow.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/intercept-events-flow.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/intercept-events-flow.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set intercept/profile-intercept.xml'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/intercept/profile-intercept.xml'
+    dest: '{{ shib_idp.home }}/conf/intercept/profile-intercept.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set credentials/secrets.properties'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/credentials/secrets.properties'
+    dest: '{{ shib_idp.home }}/credentials/secrets.properties'
+    owner: root
+    group: jetty
+    mode: 0640
+
+- name: 'Set eduGAIN attribute filter'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/edugain-attribute-filter.xml'
+    dest: '{{ shib_idp.home }}/conf/edugain-attribute-filter.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
+
+- name: 'Set federation attribute filter'
+  template:
+    src: 'assets/{{inventory_hostname}}/idp/conf/metadata-based-attribute-filter.xml'
+    dest: '{{ shib_idp.home }}/conf/metadata-based-attribute-filter.xml'
+    owner: root
+    group: jetty
+    mode: 0640
+    backup: yes
```

Comment:
These ansible tasks could all be squashed into one by the use of `with_items`.

Reply:
Most are already in the "Check files exist before fixing file permissions" which then drives the "Fix IdP file permissions" task.

Will review and remove unnecessary duplication.